File: code/generate_strategies/IR(CoT)/cpp/cpp/function_missingno_dendrogram.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <stdexcept>
#include <string>
#include "csv.h"

class MissingnoDendrogram {
public:
    std::string create_dendrogram(const std::string& data_str) {
        try {
       
            const std::string temp_file = "temp.csv";
            std::ofstream out(temp_file);
            if (!out.is_open()) {
                throw std::runtime_error("Failed to create temporary file");
            }
            out << data_str;
            out.close();

          
            io::CSVReader<3> csv_reader(temp_file);
            csv_reader.read_header(io::ignore_extra_column, "A", "B", "C");

            std::vector<std::vector<std::string>> data;
            std::vector<std::string> row(3);

            while (csv_reader.read_row(row[0], row[1], row[2])) {
                data.push_back(row);
            }

            if (data.empty()) {
                throw std::invalid_argument("Empty data");
            }

          
            for (const auto& r : data) {
                for (const auto& cell : r) {
                    if (cell.empty()) {
                        throw std::invalid_argument("Incomplete data");
                    }
                }
            }
         
            std::remove(temp_file.c_str());

            return "Dendrogram created successfully";
        } catch (const std::exception& e) {
            return "failed";
        }
    }
};

File: code/generate_strategies/IR(CoT)/java/java/function_missingno_dendrogram.java
package com.example;

import com.opencsv.CSVReader;
import java.io.StringReader;
import java.util.List;

public class MissingnoDendrogram {
    
    public String create_dendrogram(String data_str) {
        try {
            if (data_str == null || data_str.trim().isEmpty()) {
                throw new Exception("Empty input");
            }
            CSVReader reader = new CSVReader(new StringReader(data_str));
            List<String[]> rows = reader.readAll();
            
            if (rows.size() < 2) {
                throw new Exception("Not enough rows");
            }
            int headerLength = rows.get(0).length;
            
            if (headerLength < 1) {
                throw new Exception("Invalid header");
            }
            
            for (int i = 1; i < rows.size(); i++) {
                if (rows.get(i).length != headerLength) {
                    throw new Exception("Row length mismatch");
                }
            }
            
            return "Dendrogram created successfully";
        } catch (Exception e) {
            return "failed";
        }
    }
}